tests/_plugins/ui/_impl/test_input.py

```diff
@@ -115,44 +116,24 @@ def test_slider_exclusive_args() -> None:
     assert "Missing arguments" in str(e.value)
 
 
-def test_slider_out_of_bounds() -> None:
-    with pytest.raises(ValueError) as e:
-        ui.slider(1, 10, value=11)
-
-    assert "out of bounds" in str(e.value)
-
-    with pytest.raises(ValueError) as e:
-        ui.slider(1, 10, value=0)
-
-    assert "out of bounds" in str(e.value)
-
-
 def test_range_slider_init() -> None:
     slider = ui.range_slider(1, 10)
     assert slider.start == 1
     assert slider.stop == 10
-    assert slider.step is None
+    assert slider.step == 1
     assert slider.value == [1, 10]
-    assert slider.steps is None
+    assert slider.steps == [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
 
     slider = ui.range_slider(1, 10, value=[2, 5])
     assert slider.start == 1
     assert slider.stop == 10
-    assert slider.step is None
+    assert slider.step == 1
     assert slider.value == [2, 5]
-    assert slider.steps is None
+    assert slider.steps == [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
```

Comment:
I can see how this could clean up the code, but this also is much less performant. For example, if we want a range from 1-10000000 but step 1, we have to pass about more data than just the start/stop. 

I'd rather keep start/stop if we can.

Reply:
The render of the slider is really slow